pkg/querier/querier.go

```diff
@@ -16,8 +16,12 @@ package querier
 
 import (
 	"antrea.io/theia/pkg/apis/crd/v1alpha1"
+	intelligence "antrea.io/theia/pkg/apis/intelligence/v1alpha1"
 )
 
 type NPRecommendationQuerier interface {
-	GetNetworkPolicyRecommendation(namespace, name string) (*v1alpha1.NetworkPolicyRecommendation, error)
+	GetNetworkPolicyRecommendation(name string) (*intelligence.NetworkPolicyRecommendation, error)
+	CreateNetworkPolicyRecommendation(*v1alpha1.NetworkPolicyRecommendation) (*v1alpha1.NetworkPolicyRecommendation, error)
+	DeleteNetworkPolicyRecommendation(name string) error
+	ListNetworkPolicyRecommendation() (*intelligence.NetworkPolicyRecommendationList, error)
```

Comment:
Thanks Yunyang for working on this and opening this PR early. I know you might still work on this PR, but I'm not sure if you plan to change this. Anyway, I think it might be helpful if we could reach consensus on this part before integration.

The definitions here look a little bit strange to me. IMO, the controller we implement should be a general controller, which does not have dependencies on intelligence apis or a specific namespace. Any work like reading results from database or deleting results should be handled by the handlers instead of the querier functions.

I would prefer the `NPRecommendationQuerier` interface defined as follows.
```golang
type NPRecommendationQuerier interface {
	GetNetworkPolicyRecommendation(namespace, name string) (*v1alpha1.NetworkPolicyRecommendation, error)
	ListNetworkPolicyRecommendation(namespace string) ([]*v1alpha1.NetworkPolicyRecommendation, error)
	DeleteNetworkPolicyRecommendation(namespace, name string) error
	CreateNetworkPolicyRecommendation(namespace string, networkPolicyRecommendation *v1alpha1.NetworkPolicyRecommendation) (*v1alpha1.NetworkPolicyRecommendation, error)
}
```

With a very simple implementation in controller as

```golang
func (c *NPRecommendationController) GetNetworkPolicyRecommendation(namespace, name string) (*crdv1alpha1.NetworkPolicyRecommendation, error) {
	return c.npRecommendationLister.NetworkPolicyRecommendations(namespace).Get(name)
}

func (c *NPRecommendationController) ListNetworkPolicyRecommendation(namespace string) ([]*crdv1alpha1.NetworkPolicyRecommendation, error) {
	return c.npRecommendationLister.NetworkPolicyRecommendations(namespace).List(labels.Everything())
}

func (c *NPRecommendationController) DeleteNetworkPolicyRecommendation(namespace, name string) error {
	return c.crdClient.CrdV1alpha1().NetworkPolicyRecommendations(namespace).Delete(context.TODO(), name, metav1.DeleteOptions{})
}

func (c *NPRecommendationController) CreateNetworkPolicyRecommendation(namespace string, networkPolicyRecommendation *crdv1alpha1.NetworkPolicyRecommendation) (*crdv1alpha1.NetworkPolicyRecommendation, error) {
	return c.crdClient.CrdV1alpha1().NetworkPolicyRecommendations(namespace).Create(context.TODO(), networkPolicyRecommendation, metav1.CreateOptions{})
}
```
Or we could also include some generic options when defining the Interface as the followings. But I'm not sure if this is necessary as I have not come up with a use case of this based on our current requirements. 

```golang
type NPRecommendationQuerier interface {
	GetNetworkPolicyRecommendation(namespace, name string) (*v1alpha1.NetworkPolicyRecommendation, error)
	ListNetworkPolicyRecommendation(namespace string, selector labels.Selector) ([]*v1alpha1.NetworkPolicyRecommendation, error)
	DeleteNetworkPolicyRecommendation(namespace, name string, opts metav1.DeleteOptions) error
	CreateNetworkPolicyRecommendation(namespace string, networkPolicyRecommendation *v1alpha1.NetworkPolicyRecommendation, opts metav1.CreateOptions) (*v1alpha1.NetworkPolicyRecommendation, error)
}
```

Would also like to include @wsquan171 in this discussion to help us reach consensus on the `NPRecommendationQuerier` Interface definition. And I'm not quite sure if it is better to include these in PR #109

Reply:
Thanks Yanjun for reviewing.
This PR is not finished yet. I was planning to open this and have some discussion with you and Shawn.
Your suggestions make sense to me. I also think update the status in the controller is not a good idea.

I did that because I was thinking our controller might be the place to update the Status field (SparkApplication status sync, Update RecommendNP). Since the NetworkPolicyRecommendation struct is different in intelligence api and crd, I think the controller might have the resource we need for NetworkPolicyRecommendationStatus.

I'm not sure if the handler will do duplicate job if it need to access the status data from SparkApplication and ClickHouse. Also, I'm curious that, since the Status field includes a recommendedNP field, do I still need to fetch this data from ClickHouse?